docs/ballooning.md

```diff
@@ -0,0 +1,257 @@
+disables the virtio balloon statistics and otherwise represents the interval
+of time in seconds at which the balloon statistics are updated.
+
+## Security disclaimer
+
+> [!IMPORTANT]
+> The balloon device is a paravirtualized virtio device that requires cooperation
+from a driver in the guest.
+
+In normal conditions, the balloon device will:
+* not change the target size, which is set directly by the host
+* consume exactly as many pages as required to achieve the target size
+* correctly update the value of the actual size of the balloon seen by the host
+* not use pages that were previously inflated if they were not returned to the
+guest via a deflate operation (unless the `deflate_on_oom` flag was set and the
+guest is in an out of memory state)
+* provide correct statistics when available
+
+However, Firecracker does not and cannot introspect into the guest to check the
+integrity of the balloon driver. As the guest is not trusted, if the driver in
+the guest becomes compromised, the above statements are
+**no longer guaranteed**.
+
+This means that even though users use the balloon to impose restrictions on
+memory usage, they can be broken by a compromised driver in the guest. The
+balloon device operates on a best effort model and users should always ensure
+the host is prepared to handle a situation in which the Firecracker process
+uses all of the memory it was given at boot even if the balloon was used to
+restrict the amount of memory available to the guest.
+
+Users should also never rely solely on the statistics provided by the balloon
+when controlling the Firecracker process as they are provided directly by the
+guest driver.
+
+Please note that even in the case where the driver is not working properly,
+the balloon will never leak memory from one Firecracker process to another,
+nor can a guest within Firecracker access information in memory outside its
+own guest memory. In other words, memory cannot leak in or out of Firecracker
+if the driver becomes corrupted. This is guaranteed by the fact that the page
+frame numbers coming from the driver are checked to be inside the guest
+memory, then `madvise`d with the `MADV_DONTNEED` flag, which breaks the
+mappings between host physical memory (where the information is ultimately
+stored) and Firecracker virtual memory, which is what Firecracker uses to
+build the guest memory. On subsequent accesses on previously `madvise`d
+memory addresses, the memory is zeroed. Furthermore, the guest memory is
+`mmap`ped with the `MAP_PRIVATE` and `MAP_ANONYMOUS` flags, which ensure that
+even if a Firecracker yields some information through an inflate and that
+same physical page containing the information is mapped onto another
+Firecracker process, reads on that address space will see zeroes.
+
+## Prerequisites
+
+To support memory ballooning, you must use a kernel that has the memory
+ballooning driver installed (on Linux 4.14.7, the relevant settings are
```

Comment:
Why specifically `4.14.7`?

Reply:
I changed it to `4.14.193` since that is the config that we use in our tests.